### ISSUE TITLE
Reasoning for having a separate UserProfile model

### ISSUE THREAD

**Issue by noahbkim** [lgoodridge/django-uniauth#6]
Hey all, as I was reading through the project, I was wondering what benefits having a distinct `UserProfile` object (that doesn't have any of its own columns) provides over just fk'ing the `User` for all of the attached accounts and emails. Doesn't the added indirection incur some overhead, especially at scale?

Nothing to implement here, this can be tagged as question!

**Comment by lgoodridge** [comment]
Hi! I'm not sure how I missed this question for this long, but better late then never!(?)

You're correct that FKing to a separate, essentially empty, model instead of using the `User `directly introduces a little overhead, for arguably little tangible benefit. The `UserProfile` model was primarily intended to be a boundary separating all of the Uniauth models and work from the rest of the application, keeping the package self contained and only exposed to the outside world through a single model. This is nice because:

- The package supports custom user models, so it's nice to only have to deal with that once in a dedicated model, than need to remember it for everything that needs to be attached to a User-like model.
- It becomes a natural place to attach additional user-specific fields or utils, and is easily accessible via `user.user_profile.<bla>`. Right now, there's only `get_display_id()`, but I was future proofing a bit.
- It makes for a less messy breakup if you ever decide to remove the package for your app.

TLDR, it keeps the package cleanly separated from everything else and future proofs a bit for later features. Whether those pros justify the overhead as it _currently_ stands is another story however.
